gift_compassion/models/contracts.py

```diff
@@ -47,10 +47,26 @@ def invoice_paid(self, invoice):
             if invl.product_id.categ_name == GIFT_CATEGORY and \
                     invl.contract_id.child_id:
                 # Create the Sponsorship Gift
-                self.env['sponsorship.gift'].create_from_invoice_line(invl)
+                gift = self.env['sponsorship.gift']\
+                    .create_from_invoice_line(invl)
+
+                if not invl.contract_id.is_active:
+                    gift.state = 'verify'
 
         super(SponsorshipContract, self).invoice_paid(invoice)
 
+    @api.multi
+    @api.depends('activation_date', 'state')
+    def _compute_active(self):
```

Comment:
No, it's still not good, because _compute_active is a compute method which purpose is to set a value of a field. You cannot change other values from here, it's a dangerous behaviour that can do unpredictable results. You should instead inherit the `contract_active` [method found here](https://github.com/compassionch/compassion-modules/blob/770ae448805b38326f82d5f5898fc02389f81b20/sponsorship_compassion/models/contracts.py#L673).